include/base_layer.h

```diff
@@ -103,6 +103,9 @@ class BaseLayer {
                           BaseTempStates &temp_states,
                           bool state_udapte = true);
 
+    virtual void smoother(std::string next_layer_type,
+                          BaseTempStates &temp_states);
+
```

Comment:
I am not sure if the smoother function will be used for other layer like Linear, Conv2d. If it is something specific to LSTM layer, you should place it in the LSTM header (lstm_layer.h)

Reply:
I deleted it from base_layer